### DISCUSSION TITLE
18. July 2023

### DISCUSSION THREAD

**Discussion by dimakuv** [gramineproject/gramine#1462] [Meeting notes]
### Agenda

*(please write your proposed agenda items in comments under this discussion)*

- [x] Dmitrii: Gramine, GSC and Meson 1.2.0. See [this issue](https://github.com/gramineproject/gramine/issues/1459).
- [x] Dmitrii (customer ask): Why does acquiring locks takes time in an uncontended environment in `gramine-direct`?
- [x] Dmitrii: https://github.com/gramineproject/gramine/pull/1457 -- what to do?
- [x] @woju: update on RHEL situation (10 min, low priority, probably not actionable at this point)

### Woju: update on RHEL situation

- AlmaLinux update -- no bug-for-bug compatibility to RHEL, but binary compatibility (means: RHEL may choose to keep the bug open, to support apps that abuse the bug; AlmaLinux may differ in its decisions). Since Gramine is based and tested on AlmaLinux, we may become incompatible with RHEL.
- RockyLinux -- still intend to get sources from RHEL directly, by downloading an image from RHEL (and thus have access to sources inside of it). The exception is the Linux kernel, and for this they also do some trick to get the Linux kernel sources.
- OpenSUSE plan -- intend to fork RHEL. Intends to cooperate with CIQ (same people who work on RockyLinux).
- Oracle plan -- just continue their Oracle Linux.

The situation is still dynamic. Hopefully people will end up with some de-facto standard Enterprise Linux distro before Gramine v1.6.

### Dmitrii: Gramine, GSC and Meson 1.2.0

Woju, Michal: instead of fixing documentation and core Gramine stuff, we simply wait for Meson authors to reply.

See https://github.com/gramineproject/gramine/issues/1459 and https://github.com/mesonbuild/meson/issues/12006.

### Dmitrii: Issue of expanding `__FILE_NAME__`

See PR #1457

Woju: we are not in a position to compile with newer GCC (GCC 12). This is because we'll have to compile packages on newer distro, and this may have backward compatibility problems with older distros (e.g. Ubuntu 22.04).

Michal: tried to find a static macro to extract basename of a file.

Michal: modify _log() function into 4 different macros. ~~TODO: Dmitrii will do that.~~

Better way: replace the while loop with cascade of IFs, mark as always-inline and pray that the compiler optimizes it all away. TODO: Dmitrii will do that.

### Dmitrii (customer ask): Why does acquiring locks takes time in an uncontended environment in `gramine-direct`

Don and Michal say 5-10% overhead on a microbenchmark is normal.

**Comment by woju** [comment]
- [ ] @woju: update on RHEL situation (10 min, low priority, probably not actionable at this point)
